centreon/www/class/centreonDowntime.class.php

```diff
@@ -265,211 +265,319 @@ public function getInfos($id)
+            dtp.dtp_duration,
+            h.host_id,
+            h.host_name,
+            NULL as service_id,
+            NULL as service_description
+        FROM downtime_period dtp,
+            downtime dt,
+            downtime_hostgroup_relation dhr,
+            host h,
+            hostgroup_relation hgr,
+            hostgroup hg
+        WHERE dtp.dt_id = dhr.dt_id
+            AND dtp.dt_id = dt.dt_id
+            AND dhr.hg_hg_id = hgr.hostgroup_hg_id
+            AND hgr.host_host_id = h.host_id
+            AND hgr.hostgroup_hg_id = hg.hg_id
+            AND hg.hg_activate = '1'";
+
+        foreach ($this->db->query($request, \PDO::FETCH_ASSOC) as $record) {
+            $downtimes[] = $record;
         }
 
-        return $hostgroupDowntimes;
+        return $downtimes;
     }
 
-    public function getServicegroupDowntimes()
+    public function getForEnabledServicegroups()
     {
-        $servicegroupDowntimes = array();
-
-        $query = 'SELECT dt.dt_id, dt.dt_activate, dtp.dtp_start_time, dtp.dtp_end_time, dtp.dtp_day_of_week, '
-            . 'dtp.dtp_month_cycle, dtp.dtp_day_of_month, dtp.dtp_fixed, dtp.dtp_duration, '
-            . 'h.host_id, h.host_name, s.service_id, s.service_description '
-            . 'FROM downtime_period dtp, downtime dt, '
-            . 'downtime_servicegroup_relation dtr, servicegroup_relation sgr, '
-            . 'service s, host h '
-            . 'WHERE dtp.dt_id = dtr.dt_id '
-            . 'AND dtp.dt_id = dt.dt_id '
-            . 'AND dtr.sg_sg_id = sgr.servicegroup_sg_id '
-            . 'AND sgr.host_host_id = h.host_id '
-            . 'AND sgr.service_service_id = s.service_id '
-            . 'UNION DISTINCT '
-            . 'SELECT dt.dt_id, dt.dt_activate, dtp.dtp_start_time, dtp.dtp_end_time, dtp.dtp_day_of_week, '
-            . 'dtp.dtp_month_cycle, dtp.dtp_day_of_month, dtp.dtp_fixed, dtp.dtp_duration, '
-            . 'h.host_id, h.host_name, s.service_id, s.service_description '
-            . 'FROM downtime_period dtp, downtime dt, '
-            . 'downtime_servicegroup_relation dtr, '
-            . 'host_service_relation hsr, hostgroup_relation hgr, '
-            . 'service s, host h, servicegroup_relation sgr '
-            . 'WHERE dtp.dt_id = dtr.dt_id '
-            . 'AND dtp.dt_id = dt.dt_id '
-            . 'AND dtr.sg_sg_id = sgr.servicegroup_sg_id '
-            . 'AND sgr.hostgroup_hg_id IS NOT NULL '
-            . 'AND sgr.hostgroup_hg_id = hsr.hostgroup_hg_id '
-            . 'AND hsr.service_service_id = s.service_id '
-            . 'AND hsr.hostgroup_hg_id = hgr.hostgroup_hg_id '
-            . 'AND hgr.host_host_id = h.host_id ';
-
-        try {
-            $res = $this->db->query($query);
-            while ($row = $res->fetch()) {
-                $servicegroupDowntimes[] = $row;
-            }
-        } catch (\PDOException $e) {
-            // Nothing to do
+        $downtimes = [];
+
+        $request = "SELECT dt.dt_id,
```

Comment:
same as previous

Reply:
Okay I'll apply this thanks 💯

Reply:
Suggestion applied thanks for the review 👍